UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,234 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_TRANSFORM_HPP
+#define UNICORN_VIDEO_TRANSFORM_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+
+#define GLM_FORCE_RADIANS
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+/**
+ * @brief Base class for every object that can translate or rotate
+ */
+class Transform
+{
+public:
+    /** @brief Default constructor */
+    Transform();
+
+    /** @brief Default destructor */
+    virtual ~Transform() = default;
+
+    /**
+     * @brief Sets look at provided direction
+     *
+     * @param[in] direction new direction vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction);
+
+    /**
+     * @brief Sets look at provided direction
+     *
+     * @param[in] direction new direction vector
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction, glm::vec3 upVector);
```

Comment:
`SetLookAtDirection`

Reply:
Also the other `LookAtDirection`

Reply:
Since it updates orientation maybe it makes more sense to call it `SetOrientation` and move it to existing `SetOrientation`?